t/43-cli-api.t

```diff
@@ -456,6 +456,11 @@ EOF
     ($stdout, $stderr, @result) = capture sub { $api->run(@params) };
     like $stderr, qr/Connection refused/, 'aborts on connection refused';
     like $stdout, qr/failed.*retrying/, 'requests are retried on error if requested';
+    $ENV{OPENQA_CLI_RETRY_FACTOR} = 2;
+    $ENV{OPENQA_CLI_RETRY_SLEEP_TIME_S} = 3;
+    @params = ('--host', 'http://localhost:123456', '--retries', 2, 'api', 'test');
+    ($stdout, $stderr, @result) = capture sub { $api->run(@params) };
+    like $stdout, qr/delay: 6; waited 3s/, 'delay is increasing exponentially';
```

Comment:
I tried that locally and that test timed out. I assume you did not mock the sleep time correctly. Tests should never designed to wait for multiple seconds in realtime.

Reply:
I assume this retries with a delay and eventually fails. Which should be fine. Except it will increase the total runtime.

I thought we had some mocking for sleep calls, but I can't find other uses of that 🤔

Reply:
Ah. I guess you just add `use Test::Mock::Time;`

Reply:
I havent managed to make it shorter. I can only reduce the values but i dont see significant improvement

Reply:
I tried to use Test::Mock::Time (at the top of the test file), but then `retry_tx` just hangs in the second iteration when doing `$tx = $client->start($tx)`.

Reply:
@b10n1k I meant "significant" as in the statistical sense, not colloquial. An individual unit test that developers would like to execute many times during development, especially with proper TDD, must not take longer then a couple of seconds. Hence we must be very careful to not unintentionally extend the runtime of tests by some seconds. Otherwise the test becomes ineffective as a developer help and developers find other, inefficient means to develop while the test then still stays and is often only executed in CI runs, likely failing, needing multiple adaptions, taking much more time until a feature can be considered complete. That's also another reason why I consider TDD really important. I am sure if you would have added this test in the beginning you would have realized that it's taking sufficiently long to be annoying already. Only then we can take a look for improvements.

So I suggest to mock whatever is necessary to provide efficient test runs. If that is not possible then we can consider the alternative to not add a test at all as the code is at least covered by already existing tests.

Reply:
Maybe we can go without the test for now.
But it would be good to find out why Test::Mock::Time isn't working as expected.

Reply:
Test was removed to move things forward. I might take another look in the first chance